end_to_end/include/testcases/device_datastream.hpp

```diff
@@ -0,0 +1,146 @@
+// (C) Copyright 2025, SECO Mind Srl
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "case.hpp"
+#include "constants/astarte_interfaces.hpp"
+
+using AstarteDeviceSdk::AstarteMessageType;
+
+namespace testcases {
+std::unique_ptr<TestCase> device_datastream_test_case() {
```

Comment:
I would remove the `test_case` from the name of this function, as it's already implied by the namespace.